lib/src/rules/await_only_futures.dart

```diff
@@ -62,7 +62,9 @@ class _Visitor extends SimpleAstVisitor<void> {
         DartTypeUtilities.extendsClass(type, 'Future', 'dart.async') ||
         DartTypeUtilities.implementsInterface(type, 'Future', 'dart.async') ||
         DartTypeUtilities.isClass(type, 'FutureOr', 'dart.async'))) {
-      rule.reportLintForToken(node.awaitKeyword);
+      rule.reportLintForToken(node.awaitKeyword,
+          errorCode: LintCode('await_only_futures',
+              'Await applied to $type, which is not a Future'));
```

Comment:
Minor changes to match the conventions we have for diagnostic messages (specifically, quoting snippets of Dart code):

"'await' applied to '$type', which is not a 'Future'."

or, if you assign the lint code to a static field (and use the named argument `arguments` to pass in the type when invoking `reportLintForToken`), then it would be:

"'await' applied to '{0}', which is not a 'Future'."

Reply:
Done.